### ISSUE TITLE
[New build - DEFAULT] 20w27b: Piston Mechanism.

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#2590]
A new mechanism was added to the game: the piston. Perfect to quickly build landing gears!

Changes
- Small piston added.
- Large piston added.
- UI: Small screen size rescale to 1080p

Bugfixes
#2563 Multiplayer issue on some target related action.
Another tentative fix for the UI issue for large screen.
#2556 Codex UI does not fit in 720p.

**Comment by tsunamayo** [comment]
Build is up! Cheers

**Comment by ZachZent** [comment]
What place is throttle control on the list of shit to dew?

**Comment by tsunamayo** [comment]
I will first redo the ship orientation, and work on combat. 
I still dont like the idea of it because it is no longer a simple boost but a completely new control scheme for the ship, so one would have to disappear. Thats just too complex.
- 6 dof
- plane like

**Comment by ProPeach** [comment]
Why not merge the two? Elite for example does this very well. Ship orientation (pitch/yaw) is controlled by the mouse, while you can still strafe left/right/up/down with the A/D and space/ctrl keys. This is a near perfect scheme as the default in my opinion, and with the ability for a player to change their keybinds like we can now it allows anyone to tune their ship to their liking.

Throttle is managed with a sliding scale 0-100%. You use W and S to incrementally move your "goal" velocity up/down the scale, and the ship will accelerate to the chosen speed as fast as it can. Throttle is *essential* on any good flight system, for any size ship doing any activity from combat dogfighting to docking

**Comment by ZachZent** [comment]
I am sorry, but that is completely incorrect and a bad choice in every sense of the word. You want players to go only at full speed or no speed? How are we supposed to control, maneuver, dock without any sort of assistance? Even if you slow down 6DOF to a crawl like the old maneuver mode, players will still want to be able to slow down. Now there are those who want the flight system to change which I partly agree, but even keeping with what we have now nothing would be lost adding throttle. You can fly slower and 6DOF slower. The methods of flight is still the same as are the advantages and disadvantages of each. We used to have this system a long time ago. Imagine a plane, car, or anything else that only has the options of top speed and no speed. The word for that is unwieldy.

Peach's/Elite's system is exactly what is needed. A single system combining the two into one and can be "modified" depending on the mass of the ship so that there isn't a super agile dreadnought

**Comment by talrey** [comment]
Throttle control is pretty important to avoid jumping hovercraft off mountains into space, at least with how their physics work right now.

I like Peach's suggestion as well. Making strafe (and rotation?) speeds decrease with throttle would give you a low-throttle "6DoF" equivalent and a high-throttle "Boost" equivalent while giving players more control over their ships.

**Comment by SovietSponge** [comment]
I'd definitely say throttle control is absolutely necessary, the flight controls as of now are unwieldy at best, having the large deadzone and no real control over speed. Any sort of precision maneueving is incredibly difficult to damn near impossible.

**Comment by tsunamayo** [comment]
So I misunderstood you, I was thinking it was throttle control only for the boost mode, not the 6-dof... From the last discussion we had it was about this.
Still not sure about this tbh, soon you will be able to do it with the input gate, if it is only needed for docking that seems a bit excessive (as it is basically automatic). For combat I just want to do another pass on it and see if we really need it.
I understand that power user are gonna wants to steer the game through more complexity, but as you know I am super diligent to try to make it more accessible.

**Comment by ProPeach** [comment]
I was suggesting getting rid of boost mode and 6DoF as separate systems, I don't see any benefit of adding additional complexity by having two separate systems when they work so much better in one complete package. 
I also think that you might be overestimating how complex this is, it's quite a basic flight system. Aim with the mouse, WASD for throttle and roll control then strafing if you need to. This is a well tried and tested flight method, it's not for power users or niche cases. If anything, separating 6DoF and boost is more complex and difficult to grasp as a flight system because it is so unconventional and weird compared to other games

**Comment by ZachZent** [comment]
Understandable. However this feature is not exactly complicated. We can "technically" hook up all thrusters to a slider to have some ability to change speed, but this is awkward and as you said, complex. Returning the simple integrated feature of thrust control is an absolute necessity. Peach said to have it use A and S. I used to have it with page up and down (I'm a WASD pilot.) Having this option is a necessary complexity, though I'd hardly call it complex at all.

> if it is only needed for docking that seems a bit excessive (as it is basically automatic)

Technically yes, but that doesn't consider the need to maneuver into position to automatically land. Or if you want to land on the ring without a landing pad. Or slowly fly though the asteroid belt without instantly whacking a rock.

> For combat I just want to do another pass on it and see if we really need it.

Join in a game of mp and try to fly a fighter. If it only has front facing guns, you will fly past the enemy ship in a matter of seconds and need to turn around. If it has a turret, all you can really do is fly around an enemy ship at predictable speeds. For larger ships the problem is still the same. And again problems with maneuvering in not completely open spaces such as the asteroid belt or whatever land generations come in the future. I don't think anyone can fly at full speed though a canyon

**Comment by Kaiser-Indrasil** [comment]
Or, instead of W and S, you can link the throttle to the mouse scroll wheel. It's one of the few things Star Citizen actually does well. It limits the absolute speed your craft can travel at, in any direction. That would help with precision docking tremendously.

Also, if you want to leave the boost / afterburner functionality - which I actually like - I'd suggest making a small change to it. Instead of toggling it with the shift key, make it work as long as Shift is pressed down - and turn it off as soon as it's released. That would make it more intuitive to use.

**Comment by ProPeach** [comment]
Kaiser makes a good point, turning the Boost "Mode" into a Boost/Afterburner "Effect" would be smart for a temporary speed boost and would bring a lot to combat gameplay. Personally I'd make it tab, but that just comes down to pilot preference with their keybinds - same as having throttle as W/A or scroll wheel, that's just something for a player to tweak on their end

**Comment by tsunamayo** [comment]
I also want to do another pass on ship agility, I will use function that will make larger ship less agile despite having the same thrust to  weight ratio as it seems to upset people somehow. That would have the effect of slowing down all ships. I might also reduce all speed for combat, we will see, but the idea is I will do a huge pass on combat and then we will see if throttle is needed.
For docking again this is not enough, power user would always be able to implement a "docking mode" with logic.
Not saying I will do it or not, I just say we shall see!

**Comment by SageThe13th** [comment]
Vehicles in real life and and in good vehicle based games don't just fight at their top speeds. Throttle is absolutely a necessary feature. Hovercraft tend to ramp off of hills and become airborne at their top speeds. It would be ideal to control this better with a throttle setting the speed you want. Like say I wanted to add a tons of thrusters to a craft for the acceleration, but didn't intended to always go the craft's top speed. I do this all the time for hovercraft because their acceleration is garbage.

The lack of throttle actually makes controlling the vehicles more complicated not simpler. Right now, to try and stay at a constant speed I either need to press W ever so so often. Letting the speed die a little before speeding up again. Or repeatedly flipping between hitting W and S. Setting the the ideal speed with a throttle based what the task requires would let me focus on what I'm doing without having to also manage my speed constantly by fiddling with W and S. It would make controlling anything hovercraft, ships or whatever else much easier.

Take for example I'm attacking another ship in a much faster ship than than the one I'm attacking. With the current lack of throttle even if pulled behind them and wanted to stay there to get free shots on their backside I'd have little choice but to fly past them at my top speed. My other options would be to stop. Not ideal. Or to constantly fiddle with W and S to try and match speed with them. Basically focusing on my speed when I need to focus on aiming.

**Comment by ultrak2k** [comment]
My take on throttle is that I'd rather have a bit more complexity then a system that's unfun and hard to fly

**Comment by Majikmonster** [comment]
Throttle control is an absolute must. Pilots shouldn't be backed into a corner where tapping the controls is the only way to maintain a desired speed. Shooting out of any hangar or off a landing pad at max throttle is rarely if ever a good idea if there is any amount of building coverage or other traffic, which means that we are forced to tap our way around everything that we can't engage autopilot for.

Any kind of maneuvering other than flying directly from point A to point B as fast as possible requires control over throttle to be effective and safe. This is especially true around other ships. It's not just for the player's QoL, it's for doing the absolute basics of almost all ship-based tasks, namely not crashing.

The merging of flight modes and afterburner suggestions are both excellent ideas that have been featured in other spaceship games as systems that work well without being overly complicated or needlessly realistic. We don't want Microsoft Flight Simulator, just a better control scheme than we have now.

**Comment by Cpt-Fortius** [comment]
Unpopular opinion, but I've been playing games since my father first got his hands on a Commodore 64 ( pixel-art Formula-1 racer on a glorious orange-black monochrome display) and pushing and releasing W when I want to adjust or maintain my speed never seemed like a problem.

In fact, it'd be more unnatural if I was pressing W 'till the keyboard creaks and the ship is still not accelerating because I've left it in low gear. Plus, I really don't need to add one more button to the control layout.

Heck, already you can fly with, or without boost; that's throttle control enough.

other:

6DOF vs Plane-like controls? I hope we'll never go Starmade-like strafing at full speed kind of silliness. That's a massive misunderstanding of how space flight works; You may indeed have, or be assumed to have several manouvering thrusters in all directions, but they are not as powerful as your much larger main, rear-facing thrusters, thus would never be sufficient for more than minor course corrections and docking manouvers.
(IRL space flight is "accelerate forward, turn ship whichever direction you want, keep drifting on your original vector" which is a complete pain to fly in videogames.)

What strafing abilities our ships seem to have at the moment seem perfect. The only grief I have with starship controls is how suddenly they can turn a lot more than I wanted to if I move the mouse with anything but the most gentle nudge.

**Comment by esauftgz** [comment]
I believe trying to achieve the best possible controls would be a huge boon in increasing the appeal of the game to new and existing players alike. Currently, in my opinion, Elite Dangerous' flight model is definitely the best in terms of spaceship control.

Throttle allows for better combat controls and control of speed, while also reducing the need to have to constantly switch between holding the accelerator button and having to rhythmically tap it to maintain your desired speed. Therefore, I do think that it's an important component if Starship EVO would need to evolve beyond just a building game and into a space sim game.

**Comment by Majikmonster** [comment]
> Unpopular opinion, but I've been playing games since my father first got his hands on a Commodore 64 ( pixel-art Formula-1 racer on a glorious orange-black monochrome display) and pushing and releasing W when I want to adjust or maintain my speed never seemed like a problem.

You got used to a subpar system of control. While there's nothing wrong with mastering bad controls, that doesn't mean we should stick with or even encourage that kind of control scheme. Tsuna can do better than an ancient game made for an ancient 
computer system.

> In fact, it'd be more unnatural if I was pressing W 'till the keyboard creaks and the ship is still not accelerating because I've left it in low gear. Plus, I really don't need to add one more button to the control layout.

Slamming the gas on gear 1 of your transmission will never result in you going past gear 1, because that's how it works. It's how all sufficiently advanced cars, planes, trains, and boats control throttle. Each has a gearbox or electronic control of some kind that limits speed and/or acceleration. Extending that to spaceships isn't even remotely a stretch from natural. Any modern space sim includes some method of controlling max speed/acceleration, even GUI simulator SpaceEngineers does it (badly).

> Heck, already you can fly with, or without boost; that's throttle control enough.

 Toggle between fast and faster? I don't think your definition of control fits what the rest of us are referencing.

A good quality flight control scheme will always have throttle control. This is one of many reasons why No Man's Sky is a mediocre spaceship game, and one of many reasons why Starmade flight control sucks as horribly as it does. Neither have real **control**, only "press button=go fast" and "hold button=go faster", which is laughable even compared to aircraft of the mid 1920s which ditched throttle-less rotary engines (that needed you to cut off fuel to slow down) for much better ones.

TL;DR throttle is a requirement for good vehicle games (not including overly retro or child-oriented games)

**Comment by SovietSponge** [comment]
I very much agree that throttle is an absolute necessity, no game with decent controls is without it and though the flight controls have been massively improved they'll never really be much good if you can only sit still or go so damn fast you can't even realise what's going on around you. And good god please add it so we don't have to mash w to keep our desired speed, my keyboard can't take it, my fingers can't take it and my brain can't take it.

**Comment by Rexxios** [comment]
Just think about it, Cars can go over 100mph (160km/h), but how often do they go top speed?

**Comment by tsunamayo** [comment]
Does Avorion / Space Engineers have throttle control?

**Comment by ProPeach** [comment]
Space Engineers doesn't, but it operates on a strictly Newtonian flight model with 6DoF only and a need for thrusters in 6 directions, so I don't think its flight model is comparable to the more arcadey model that I think you're going for here. In that game you can disable your inertial dampeners to cruise at your current velocity with your fingers off the keyboard though which is a kind of pseudo throttle (with inertial dampeners on, the ships thrusters will attempt to bring it to a stop). It also has a ~110m/s hard limited max speed for all ships, so again it's pretty different.

Avorion doesn't either (I haven't played it so someone please correct me if I'm wrong), but just has a very simple mouse aim and press W to go forward system with no roll/yaw/pitch control so doesn't have much crossover to the dogfighting gameplay that you seem to be aiming for.

**Comment by tsunamayo** [comment]
@ProPeach you can roll in avorion. A lot of people requested this kind of control. 
Also on what you are requesting: you where saying map throttle to W ans S, but that would mean the ship always move forward or your still have to press a move forward key? What is doing elite in that regard? 

Also what is doing elite for selecting target? I think with the current mini deadzone it is near impossible to select a distance target comfortably. Ships turn too fast, I think middle mouse is just the wrong way of doing this, I might switch to a button based selection that would toggle between target in front of the ship.

**Comment by ProPeach** [comment]
Ah gotcha, thanks for clarifying!
I was suggesting the same system as Elite, to where you have a throttle bar out of 100% on your HUD that can be moved with W and S (or other keys/scroll wheel if you want to rebind them). If you don't touch W or S, the throttle level doesn't change. So you can hold W to increase throttle to say 80%, then leave it there at 80% until you want to change it again. This is super easy to learn and very comfortable to use. Using that system you also have a default keybind on the X key that instantly sets the throttle at 0% so your ship makes a quick stop should you need it.

In Elite you press T the ship you're looking at which is also rebindable, plus a few hotkeys for things like selecting the nearest hostile (VERY useful!) or cycling through the list of ships in your area. They have middle mouse button as the freelook key so you can easily look around your cockpit and target a ship without moving your ship. I think this just comes down to personal preference to be honest, the only real differences between the two targeting methods are the hotkeys and that SEVO ships turn much faster which I think you mentioned would be addressed with an upcoming mass re-balance? They're definitely too twitchy at the moment I agree.

**Comment by tsunamayo** [comment]
yeah but are you moving forward all the time or do you have to press a key to move forward? W and S are the move forward and backward in the current setup.

**Comment by Kaiser-Indrasil** [comment]
You press W and S only to achieve your desired speed - the ship maintains it.

**Comment by ProPeach** [comment]
The ship is at 0% throttle unless you increase it to go forward, it's not like the current boost mode where you move forward constantly with no control like you're stuck at 100% throttle.

**Comment by ProPeach** [comment]
[Elite Dangerous Throttle Clip.zip](https://github.com/tsunamayo/Starship-EVO/files/4920697/Elite.Dangerous.Throttle.Clip.zip)

Here's a video just to make sure we're on the same page as it's easier to show, watch the throttle bar to the right side of the main console.
I start at 0m/s, then use w to increase my speed to ~60m/s. The ship holds that speed. I hold down the S key to move the throttle level down and the ship follows, then use W again to increase it maximum. 
I then hit X to set the throttle at 0 and the ship slows down.

The throttle level effectively sets a "goal" for the ships velocity, and then the ship tries to meet that goal as quickly as possible considering the agility of the ship.

**Comment by tsunamayo** [comment]
@ProPeach thanks! So my problem with this is that you basically cant maneuver, what about mining an asteroid or salvaging a wreck? It make no sense for the game I want to do. Or maybe with just boost mode, but your are not happy with that and my neither. Have you played Everspace? Thats the kind of control I want. 

I will also have to widen the deadzone, as it is impossible to select any small target with an agile ship, and I constantly find myself overshooting when correcting back to center point. 
I am not sure to understand why people dont like wide deadzone. It work perfectly in Everspace and it is quite wide. I think it is again a case of misunderstanding the initial issue: issue was the physical model (which of course was hard for the player to understand), not the deadzone.
But first I will nerf agility, ships just turn too fast, and I will nerf it even more for large ship which was something everybody was requesting.

**Comment by ZachZent** [comment]
I have not played Everspace, but watching the combat shows a few key fundamental differences between it and SEVO. The biggest difference being that ships are much larger. At least visually. Dog fighting seems to be at a general range of 200m, but visually it is much closer.

![](https://cdn.discordapp.com/attachments/271723356795961345/736844724148764722/unknown.png)
Because of this visual range, the player doesn't need to do hard turns as often to pursue a fighter. It is more about small adjustments. Hence the large deadzone.

![](https://cdn.discordapp.com/attachments/271723356795961345/736845785190105118/unknown.png)
This is the same distance in SEVO looking at a generally standard size fighter. About 10m long and 6m wingtip to wingtip. You can hardly see it.

![](https://cdn.discordapp.com/attachments/406664114249793536/736846701708574740/unknown.png)
This is a 50 meter ship 200m away. Visually it looks about the same as the Everspace picture, but if you need 40 extra meters just to reasonably see a ship, fighter to fighter combat is not going to happen like Everspace.

_______________________________________

Agility is also a big difference in SEVO versus Everspace. Ships appear much slower in Everspace where you are able to actually pursue someone from a short, well "short" distance. This allows for more controllable combat. In SEVO right now, you need turrets if you want to hit anything, and I mean anything. If you have weapons pointed forward, best you are getting is a glancing shot. And as an aside, this is one of the reasons I'm for increased fire rate for the non-gattling weapons. One shot every second where you can only get a glancing shot if lucky means your potential DPS will garbage. Adding the limited gimbal to weapons would be great, but currently it would still be too difficult to hit anyone. Much less hit them repeatably to death.

Back to topic, turn rate is also much more controllable, which you are aware of. My two cents, there should be a max turn speed depending on the mass of the ship rather than thruster power exclusively. You can get lower then it by putting fewer maneuvering thrusters, but not higher even if you put down more. This will give ships a more manageable and predictable turn rate: fighters turn faster, larger ships slower. And placing down extras is still useful as if some are destroyed there are backups.

![](https://cdn.discordapp.com/attachments/406664114249793536/736850268313878578/unknown.png)
Obviously the # Of Thrusters would be total thrust force, but you get the idea.

_____________________

Finally back to the main topic, throttle. From what I see in Everspace, there are three flight "modes." Normal flight, an airbreak, and a short burst of speed. I see this working in Everspace, but it will not work here. Everspace is only fighter vs fighter combat (unless I missed something) and the fighters have predictable speeds determined by the developer and not the players. You don't need throttle in that game as everyone flies the same speed. Breaking and boosting is a combat tactic rather than a dedicated movement mode (There is no combat without movement - Ender Wiggin.) Unless you plan to give all ships the same speed, some ability to control your throttle is needed.

I know that W and S are out of the picture unless you want to completely redo the movement system. The best alternative I can see is using the scroll wheel. Everyone regardless of control system uses the mouse (I fly with WASD rather than the mouse.) One scroll wheel click up, throttle goes up 10%. One down and it goes down 10%. This is not complicated at all for anyone and is easy enough to control that you can go from 100% to 0% easily. With the exception of trackpad players, but those people need to EVOlove (hahahahaha.)

**Comment by tsunamayo** [comment]
How do you play with WASD without the mouse?
Now if I understand the problem of the deadzone is just to aim at stuff? Again this is why I am picky and I try to ask detailed feedback, because otherwise I am fixing it the wrong way.
Smaller deadzone => no mouse select possible. Elite dont use mouse select.
Now for the aim issue the answer is simple and it is gimbal, not a smaller deadzone. Everspace has a massive deadzone, or either a power curve (move super slow in the middle), but it has gimbal, so you can very much shoot at stuff easily. I also think it might have a small auto-aim like elite. Elite also has gimballed weapon, even with a tighter deadzone it is gonna be essential.
Also agility is getting a nerf in next patch, I actually just did what you suggested just before reading it lol. Bigger ship will turn slower. I also think I will cap total agility at 100% like you suggested.

**Comment by tsunamayo** [comment]
Also do you play with mouse recenter in elite? I tried and it was just unplayable, but maybe I had wrong settings. You constantly needs to move your mouse to turn, it was really painful and unnatural.
Scrollwheel could work and is the only acceptable mapping I see, but that means not using it for any kind of weapon / action selection (which I am fine with)

**Comment by Kaiser-Indrasil** [comment]
I agree that the throttle is needed. Even fighters, with their speed and maneouvreability, need to land or dock up eventually, which requires precision that throttle would provide. With connectors introduced recently, players will build detachable cargo crates and carrier fighters and they will want not to jerk around at full speed around the hangar or accidentally bump the cargo crate kilometers away.

Also, using scroll wheel for throttle sounds like the best option. Star Citizen does it this way and it's very convenient.
For weapon selection I feel like the numerical keys like 1-9 are better suited since I doubt anyone would more than 9 or 10 different weapons groups. Usually it's 1 or 2, and rarely more than 3.

On top of that, I suggest that the throttle increments be logarythmic since players will need to fine-tune their precision movements at slow speeds but at high speeds the difference between 200 and 201 m/s is pretty much unnoticeable.

**Comment by ZachZent** [comment]
WS for pitch, AD for yaw. QE for roll. Mouse for camera and weapons. Num pad for maneuvering. There are advantages to this for me, better control on the flight and can look independently from flying at all times. Downside being can't maneuver and fly at the same time unless I want to only look forward.

Peach and the others are better at explaining the specifics, I'm just mentioning the main differences between Everspace and SEVO combined with testing stuff in mp and why a direct copy would have its problems. Especially on the visual side being that there is no way in Everspace that the pic taken is 200m.

The aim issue is a bit more than the deadspace and gimbal, but it is a start.

Finally for the scroll wheel, you can still click it to select targets. Unless you are using this mouse specifically, you should be able to click the mouse and scroll without interfering with the other.
https://cdn.discordapp.com/attachments/406664114249793536/736860747258396742/shopping.png

__________________________

Speaking to Indrasil, I don't think a weapon selection like Everspace is really possible. Unless you want one barrel only that can change between modes. I did speak about a similar thing awhile back, it may relate to what you are looking for or not. https://github.com/tsunamayo/Starship-EVO/issues/2089

**Comment by AlienXtream** [comment]
why not have multiple control schemes? everyone likes something different so having a couple options in the ship control block would be nice IMO. personally i like the idea that i have to KEEP moving my mouse to turn as its more similar to how you control the camera in other cases. this would also perhaps mean that once modding becomes possible modders could add NEW control schemes. to clarify i dont mean rebinds but totally custom programed control settings, from deadzones to turn acceleration. best of all if you require servers and clients mods to be synced (i.e no client only mods) then you dont need to worry about players having an "unfair advantage" because apart from the default options available modded controls would have to be installed on the server by the owner. load a workshop ship with non standard control configurations? warn the player of the missing mod and load default controls.

**Comment by ZachZent** [comment]
- I actually just did what you suggested just before reading it lol. Bigger ship will turn slower. I also think I will cap total agility at 100% like you suggested.

Just to clarify when I was making that suggestion, the graph would be for one specific mass type. Not every mass. A larger ship/mass would have the same type of line, but different Y values. Is that the plan?

**Comment by ProPeach** [comment]
@tsunamayo what do you mean by not being able to manoeuvre? If it's about finer control like flying in a wreck or mining, I'd actually argue that throttle system is better for that, as you can just set your speed to 20% so you fly slowly and then concentrate on not flying with the mouse. It's actually *less* effort that constantly having to hold/release the W key to stay at a slow speed. Also it's important to remember the larger ships in SEVO will be using this same system - it's much easier to set your ship at 70% speed and then switch your attention to using your turrets than constantly having to think about the W key. Another great use for this is matching your speed with other ships, it's impossible to do without a throttle function like this. 

I've just played some Everspace to refresh my memory, and I do like the control scheme but I do't think it will fit your game. The reason the wide deadzone works in that game is because the weapons have a *huge* gimbal, they can basically hit anything on your screen if you aim the mouse at it. This feels nice to use, but it won't work when players can place their guns wherever they like.
It's difficult to illustrate, but here you can see a ~20 degree gimbal or so on the guns in Everspace. In SEVO, a player be limited to placing their guns on exposed positions to make sure they don't shoot themselves or have the laser clip through their ship which is ugly.
![image](https://user-images.githubusercontent.com/27857711/88476787-22a8f000-cf33-11ea-8e6f-d3a93669c58e.png)

I also don't think this is the right method for a game with PvP elements, it's very hard to actually miss the target in Everspace. Dogfights won't be fun if a gun can gimbal to anywhere on the screen to negate any skill based evasive flying.

The guns in SEVO will absolutely need some kind of gimbal as you say, you mention the fixed weapons in Elite which in my opinion is the best way to go for this - a small gimbal for small corrections and to make it a little easier to hit the target.

The problem with the deadzone before in SEVO was that it just felt bad, not that it was only a problem with aiming. It didn't feel like you had any control over where your ship was pointing. The more recent changes to the mouse controls are excellent though and make this so much better. Increasing the deadzone a little bit is ok, but as long as you introduce a setting for it so it is configurable.

Also, Elite does use mouse select. You aim with the mouse and press T to target what you're looking at, which is made a lot easier by activating the free-look mode (MMB by default I think). This is identical to what we have in SEVO, and it will work well with the lowered agility which sounds like a great plan

**Comment by tsunamayo** [comment]
The target selection in Elite is a bit different, I dont think you have to be exactly on target to select it, whereas with the mouse selection you have. Do you think the current selection is okay in SEVO? I certainly dont - it is extremely frustrating. It all come down to that very small deadzone. You basically need your ship to be perfectly aligned to the target and then make your selection.

Regarding gimbal we will see how it goes, but be it 5% or 20% you are gonna hit your ship on side weapon anyway.
For the lack of maneuverability in Elite I was just referring to the fact that in that layout you cant really move forward / backward by a just bit, it is not made for that, it is perfect for cruising not exploring your surrounding. You want to move forward by 10m? You need to open your throttle a bit, then at one point move it back to zero. In the Elite layout I am talking of course, if keep current layout and map throttle to mouse wheel it is different.
No need to pound me for the 100th time on throttle I have heard your message ;), now I am being very cautious with adding features, as I feel I wont be able to remove it without a big outcry.

**Comment by ProPeach** [comment]
I'm sorry, I didn't mean to pound you! Sometimes it feels like we're not getting across ideas clearly, or how important some things are so forgive us if we repeat ourselves a little.
Personally I use the free look button to target things so my ship doesn't try to follow my mouse, I agree that it's not ideal and your agility nerf should help a lot with that, as well as perhaps making it so that you don't have to get your mouse exactly in the target box to select a ship.
Honestly the gimbal can be super small in terms of degrees, with only a 2 degree gimbal either way that gives you a 14m grace area at 200m which would already be a huge help. Looking forward to testing this out! 

With regards to maneuverability using each system I totally get your point, it's convenient to just be able to tap the forward key for a tiny burst speed and then immediately coast to a stop. But I think this is the only situation where that system could be considered superior, for all other (more common) scenarios being able to set your speed at a constant value is much more convenient and easier on the player. Even when doing small adjustments, surely it's easier to be able to tap the increase speed button to set it to a low value and then hit a hotkey (like X) to set it back to 0 when you want to bring your ship to a dead stop. That way you can move forward just a bit while having complete control over how much a little bit actually is. Maybe this is just personal preference, but it's very powerful once you try it. It's brilliant for cruising, dogfighting, coasting through tight spaces or docking as your ship moves at a predictable, constant velocity

**Comment by tsunamayo** [comment]
No pb mate, I always appreciate your feedback.
Pb is (my fault), as there is no real gameplay you dont see the use yet, but the game will be of course about mining, salvaging ect, in short exploring. So my reference for this are much more Everspace / Avorion than elite. In elite btw as there is no 3rd person (I think?) you dont really have any spacial awareness. So for sure I dont plan at all on giving up the 6dof. I am not sure on why we are discussing that, are you suggesting I should give up 6dof to get something like Elite?

**Comment by tsunamayo** [comment]
@ZachZent this is my patch not so far:

Following players feedback a new Spaceship agility model is introduced: 
Ship with more mass will turner slower at a given agility. 
Some examples: 
Up to 20T = 100% speed.
50T => 91% speed.
100T => 79% speed.
1000T => 49% speed.
10000T => 21% speed.
You might need to put more thrusters on your larger ship if it turns too slowly!

**Comment by ProPeach** [comment]
No, I agree that 6DoF is super important for all those activities. I realise where some of the confusion may come from, Elite actually uses a hybrid system - the adjustable speed throttle applies to your forward/backwards momentum only. You set your forward speed via the throttle slider, but you can still strafe up/down/left/right and roll with the keyboard while travelling at that constant forward speed. Perhaps it should be called 4DoF instead? I should have said this sooner, but from a few of your previous replies it sounded like you had played the game so i didnt mention it.

I'm a little confused with your first statement though, Elite has all those activities in it so I'm not sure why it wouldn't be part of the comparison over games like Avorion.

**Comment by Kaiser-Indrasil** [comment]
@ProPeach 4DoF flight model (with its derivative 3DoF used in games like Freespace 2 and newer Star Wars: Battlefront) prohibits the player from strafing up, down, left, and right. In other words, airplane in space with extra backing up.

Personally, I'm more of a fan of speed limiter rather than Elite-style throttle applied to the forward-backward axis of movement. In such flight model the player sets the speed limit with their mouse scroll anywhere from 0 to 100% thrust on top of the flight model we already have in Starship Evo. Yes, it requires the player to hold down the W button constantly during cruise - something that could be remedied by introducing a Cruise button, something what Star Citizen does for example - but it's much more intuitive and unobtrusive during combat. That way, I can strafe forward and backward in an evasive maneuver. Furthermore, having a universal speed limiter instead of a forward-backward throttle comes in handy where your main axis of movement is up-down or left-right. A good example would be trying to align your port side coupling hatch to a station's dock. In this case, your 2 axes of side-strafing are up-down and forward-backward.

**Comment by Majikmonster** [comment]
The speed limit type of throttle control Kaiser is talking about is the same system I was referring to. Think transmission on a car: You still need to press the gas to accelerate, but the gear (throttle) you set determines **how fast you accelerate** and what **max speed you can reach**.

As for the deadzone, the tiny one we have now (I'm assuming it still is tiny as I haven't played the last couple updates) has been a godsend for ship flight and static weapon aiming. Making it any larger than it is would only hurt. If it's a matter of aiming weapons, then static weapons become more and more useless the larger the deadzone becomes. If it's for maneuvering, you can forget ever flying near anything or "buzzing the tower" with how botched precision steering would be. The faster (and now also smaller) your ship is, the more punishing it will be with a large deadzone as finer control disappears.

Before, a ship couldn't even dodge obstacles or avoid accidents because we had to fling the mouse across to the edge of the zone and hope the turn rate could make up for the lack of control while simultaneously not making you do flips (_when filming for the community-made game promotion cinematic, there were terrible troubles with the deadzone making it painfully difficult to do anything with the precision tasks I'be mentioned above, just ask Zach and the others who flew ships then_). Now we don't have to worry about flipping as much with the reworked agility. What we will have to worry about if the deadzone goes back to being bigger combined with this new agility patch is not being able to turn fast enough.

Tsuna, whatever system you've envisioned that needs a large deadzone isn't compatible with how the game handles scale, speed, and your desire to have less turret spam.

On top of a smaller deadzone, we need less laser inaccuracy and an overall higher rate of fire if player-aimed weapons are going to be at all useful, especially if they are static. As Zach mentioned above, turrets are currently the only viable option for hitting moving targets. That, and we have to spam weapons just to fire enough bullets to get a reasonable hit chance percentage. Otherwise a ship with a single shield module can easily slip by with little to no damage to the hull. The faster the target is, the worse the chance is, smaller target size only worsens that, and if compounded with long distances the results are pathetic at best.

**TL;DR:**
**throttle**= work like a car transmission
**deadzone**= must be tiny **or** scale with speed (faster=smaller)
**weapons**= need to be better overall if you want to avoid weapon/turret spam

P.S. I might have missed/misread something, so expect edits.

**Comment by JanZelenka** [comment]
Hi all,
I'm probably too late for the party but hope to add some valuable feedback anyway. Regarding Elite flight model, unless I missed it somewhere, I don't think people got it right here.
Elite is a full Newtonian model with 2 simplifications:
- A speed limit
- Ignores gyroscopic effect
Both I can live with perfectly.

As a pilot you do nothing else than control the thrusters. I'd better start here with how controllers are implemented and then move onto keyboard.
When you apply an axis input, the connected thruster fires. The more you apply the axis control, the stronger the thrust. Once you return your axis to idle position, thruster shuts down. The game calculates the resulting forces and converts them to the ship movement/rotation.

With keyboard controls when you hit a key, assigned thruster fires at full thrust, you release a key, thruster shuts down. With one exception here: forward/reverse thrusters. When you press the forward/reverse key, the thrust starts to increase. Once you release it, the thrust remains at the set level.

That is the core. Now what people were referring to here was what happens when you turn the _Flight Assist_ on (which in fact is on by default). Now all of the above remains true, only once you stop applying thrust in one way or another, your ship starts firing opposite thrusters until you motion/rotation is brought to a stop.

It would be really cool if Starship EVO worked the same way.
